MoltenVK/MoltenVK/GPUObjects/MVKImage.mm

```diff
@@ -1522,6 +1522,24 @@
                 setConfigurationResult(reportError(VK_ERROR_FEATURE_NOT_PRESENT, "vkCreateImageView(): 2D views on 3D images can only be used as color attachments."));
             }
         }
+
+        // If a 2D array view on a 2D image with layerCount 1, and the only usages are
+        // attachment usages, then force the use of a 2D non-arrayed view. This is important for
+        // input attachments, or they won't match the types declared in the fragment shader.
+        // Transfer usages are OK: the transfer commands don't use image views.
+        if (mvkIsOnlyAnyFlagEnabled(_usage, (VK_IMAGE_USAGE_TRANSFER_SRC_BIT |
+                                             VK_IMAGE_USAGE_TRANSFER_DST_BIT |
+                                             VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT |
+                                             VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT |
+                                             VK_IMAGE_USAGE_INPUT_ATTACHMENT_BIT |
+                                             VK_IMAGE_USAGE_TRANSIENT_ATTACHMENT_BIT))) {
+            if (_mtlTextureType == MTLTextureType2DArray && _image->_mtlTextureType == MTLTextureType2D) {
+                _mtlTextureType = MTLTextureType2D;
+            } else if (_image->_mtlTextureType == MTLTextureType2DMultisample) {
```

Comment:
For consistency with the comment and non-multisample test two lines above, should this include a test for `_mtlTextureType == MTLTextureType2DMultisampleArray` (guarded by MVK_MACOS`)?

Reply:
Maybe it should. I've made this change.